pmpro-signup-shortcode.php

```diff
@@ -381,7 +381,7 @@ function pmprosus_signup_shortcode($atts, $content=null, $code="")
 					if( !empty( $tospage ) ){
 						$tospage = get_post( $tospage );
 						?>
-						<input type="checkbox" name="tos" value="1" id="tos" /> <label class="pmpro_label-inline pmpro_clickable" for="tos"><?php printf(__('I agree to the %s', 'paid-memberships-pro' ), $tospage->post_title);?></label>
+						<input type="checkbox" name="tos" value="1" id="tos" /> <label class="pmpro_label-inline pmpro_clickable" for="tos"><?php printf( esc_html__('I agree to the %s', 'paid-memberships-pro' ), esc_html( $tospage->post_title ) );?></label>

```

Comment:
```suggestion
						<input type="checkbox" name="tos" value="1" id="tos" /> <label class="pmpro_label-inline pmpro_clickable" for="tos"><?php echo esc_html( sprintf( __( 'I agree to the %s', 'paid-memberships-pro' ), $tospage->post_title ) );?></label>

```

Reply:
Let's always shoot for one escaping function around the whole concatenated string vs escaping parts. This is usually more clear, and also handles the extreme edge case where partial changes in each string that aren't escaped individually become harmful once combined.